avocado/utils/network/hosts.py

```diff
@@ -0,0 +1,115 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2019-2020 Red Hat Inc.
+# Authors : Beraldo Leal <bleal@redhat.com>
+
+"""
+This module provides an useful API for hosts in a network.
+"""
+
+from ..ssh import Session
+
+from .common import run_command
+from .interfaces import NetworkInterface
+from .exceptions import NWException
+
+
+class Host:
+    """This class represents a base Host and shouldn't be instantiated.
+
+    Use one of the child classes (LocalHost or RemoteHost).
+
+    During the initialization of a child, all interfaces will be detected and
+    available via `interfaces` attribute. This could be accessed on LocalHost
+    and RemoteHost instances.
+
+    So, for instance, you could have a local and a remote host::
+
+        remote = RemoteHost(host='foo', port=22,
+                            username='foo', password='bar')
+        local = LocalHost()
+
+    You can iterate over the network interfaces of any host::
+
+        for i in remote.interfaces:
+            print(i.name, i.is_link_up())
+    """
+    def __init__(self, host):
+        self.host = host
+
+    @property
+    def interfaces(self):
+        cmd = 'ls /sys/class/net'
+        try:
+            names = run_command(cmd, self).split()
```

Comment:
here is self.host

Reply:
@smruti77, why? Are you sure? Did you execute the code? `run_command()` receives the command and the host object.

Reply:
@beraldoleal  yes i got it, getting error.it should be  run_command(cmd, self.host)

Reply:
?? did not understand you. Are you still getting errors? If yes, please send the commands. Also see my another comment here.